specs/logging_implementation.adoc

```diff
@@ -0,0 +1,201 @@
+  exact error message must be hardcoded by the developers. No backtraces or
+  system generated error messages should be sent as notifications to the
+  administrator.
+* Notifications sent to the administrator should provide sufficient context to
+  enable debugging of the problem via investigation into the log files.
+* Notifications and alerts need to be delivered to the node agent alert sockets.
+* Commons needs to implement a logging library that hides all the implementation
+  details and configuration options from any of the components using the library.
+  Components should be able to blindly emit messages to the Message Handler module
+  in Tendrl and it should be able to deliver them to either syslog, log files,
+  alert sockets etc. depending upon the metadata and configuration.
+* Syslog must be the default logging configuration. Logging to individual
+  configuration files should be thoroughly discouraged.
+
+== Alternatives
+
+* None
+
+== Data model impact
+
+None
+
+== Impacted modules
+
+=== Tendrl/common impact
+* Create a new class called "Message" in common for message handling with attributes
+  handler_name, message_type, message_id, log_level, message_tag, message, severity,
+  timestamp. (other attributes are filename, function_name, line_number, timestamp.
+  These details are appended with message automatically).
+```
+Message class object structure
+     log_handler_name
+     message_type(log/alert/flight_operation)
+     log_level(info/error..)
+     message
+     message_trace_back_flag(True/False)
+     message_uuid (optional for log)
+     severity(optional for log and flight_operation)
+     tag(optional for log and flight_operation)
+
+     timestamp(generated and appended by handler automatically)
+    (Before log any message filename, function_name, line_number appended with message
+    by handler).
+
+    1. For log - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag)
+    2. For flight_oper - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag, message_uuid(job_id))
+    3. For alert - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag, message-uuid(alert_id), severity, tag)
+```
+* Create a new function called message_handler in Message class and process the
+  object values.
+* If the message type is Alert then store the message in log file with uuid (log level
+  for alert is always warning). and put the object in node_agent socket.
+* If the message type is flight_operation then log the message in file with message
+  uuid (uuid is used to trace the filght operation).
+* If the type is Log then process then store message in log file.
+
+=== Tendrl/node_agent impact
+
+* Create socket to receive alert.
+* Import message class from common and change all log statement (it is handled in another PR).
```

Comment:
All I meant is instead if saying handled in another PR put the reference of that PR